src/component/component/index.js

```diff
@@ -137,19 +137,31 @@ export class Component extends BaseComponent {
         // Register all of the drivers for instantiating components. The model used is -- there's a standard javascript
         // way of rendering a component, then each other technology (e.g. react) needs to hook into that interface.
         // This makes us a little more pluggable and loosely coupled.
-
+        this.driverResults = {};
         this.registerDrivers();
         this.registerChild();
         this.listenDelegate();
     }
 
+    canBeRenderedByDriver(driverName) {
+        // currently all drivers only render in IFrame
+        try {
+            this.validateRenderContext(CONTEXT_TYPES.IFRAME);
+            return true;
+        } 
+        catch (e) {
+            return false;
+        } 
+    }
+
     registerDrivers() {
         this.driverCache = {};
 
         for (let driverName of Object.keys(drivers)) {
+            this.driverResults[driverName] = {};
             let driver = drivers[driverName];
             let glob = driver.global();
-            if (glob) {
+            if (glob && this.canBeRenderedByDriver(driverName)) {
```

Comment:
Let's just make this `this.contexts[CONTEXT_TYPES.IFRAME]` rather than going down the try/catch route.